101-bundle-json.md

```diff
@@ -816,81 +849,62 @@ The usage of extensions is undefined. However, bundles SHOULD be installable by
-    - `properties`: The set of named properties and schemas for their values included in an object type output. MUST be an object where each named property contains a JSON schema. (OPTIONAL)
-    - `propertyNames`: Output validation requiring that each property name in an object match the specified schema. MUST be a JSON schema. (OPTIONAL)
-    - `sensitive`: Indicates that a runtime should treat the output as a sensitive value. Defaults to false (OPTIONAL)
-    - `then`: Output validation requiring that the resulting value match the specified schema. Only matches if the resulting value matches the schema provided in the `if` property. MUST be a JSON schema. (OPTIONAL)
-    - `title`: Short, human-readable descriptive name for the field. Can be used to decorate a user interface. MUST be a string. (OPTIONAL)
-    - `type`: Output validation requiring that the resulting value is either a "null", "boolean", "object", "array", "number", "string", or "integer". MUST be a string or an array of strings with unique elements. (OPTIONAL)
-    - `uniqueItems`: Output validation requiring the items included in the resulting array be unique. MUST be a boolean. (OPTIONAL)
+  - `fields`: name/value pairs describing outputs:
+    - `<name>`: The name of the output. In the example above, this is `clientCert`, `hostName`, and `port`. This is mapped to a value definition, which contains the following fields (REQUIRED):
+      - `applyTo`: restricts this output to a given list of actions. If empty or missing, applies to all actions (OPTIONAL)
+      - `definition`: The name of a definition schema that is used to validate the output content. (REQUIRED)
+      - `description`: Descriptive text for the field. Can be used to decorate a user interface. MUST be a string. (OPTIONAL)
+      - `path`: The fully qualified path to a file that will be created (REQUIRED). The path specified MUST be a _strict_ subpath of `/cnab/app/outputs` and MUST be distinct from the paths for all other outputs in this bundle.
+  - `required`: A list of required outputs. MUST be an array of strings.(OPTIONAL)
 
-An invocation image should write outputs to a file specified by the `path` attribute for each output. A bundle runtime can then extract values from the specified path and present them to a user. A runtime can leverage appropriate [in-memory](https://docs.docker.com/v17.09/engine/admin/volumes/tmpfs/#choosing-the-tmpfs-or-mount-flag) volume mounted at the `path` location for storing these outputs.
 
-For more information on the supported output properties, visit the [JSON Schema documentation](https://json-schema.org/)
+An invocation image should write outputs to a file specified by the `path` attribute for each output. A bundle runtime can then extract values from the specified path and present them to a user.
```

Comment:
Depending on what we decide the `required` behaviour is, we should also update the wording here.

Reply:
I added the following line to clarify the requirement of outputs:
`All outputs that apply to a specified action are considered to be required. If an output is missing at the end of an action, the runtime should report this condition as an error to the user.`